### PR TITLE
[Feature] CreateEntry and UpdateEntry methods

### PR DESCRIPTION
**Pull request by dmitriyslaym**
https://github.com/capsulajs/configuration-service/issues/39

Release notes:
- save method is still preserved, but deprecated
- added createEntry and updateEntry public methods

### CODE REVIEW

**Review by VictoriaBordian — APPROVED** [review]
QA passed

File: tests/ConfigurationServiceScalecube/data.test.ts [hunk 1]
```diff
@@ -42,4 +42,16 @@ describe('Test suite for the ConfigurationServiceLocalScalecube', () => {
...
     expect(await configService.save({repository, key, value})).toEqual({repository});
   });
+
+  it('Call createEntry() with a new key should create a new entry', () => {
+    expect.assertions(1);
+    mockedCallback.mockResolvedValueOnce({ repository });
+    return expect(configService.createEntry({ repository, key, value })).resolves.toEqual({});
+  });
+
+  it('Call updateEntry() with an existing key should update the relevant entry', () => {
```
Comment (stephanebenayoun) [inline tests/ConfigurationServiceScalecube/data.test.ts]: What about `Call createEntry() with an existing key should ...` ?

Comment (stephanebenayoun) [reply tests/ConfigurationServiceScalecube/data.test.ts]: And `Call updateEntry() with an new key should ...(error)`

File: src/provider/LocalStorage/ConfigurationServiceLocalStorage.ts [hunk 2]
```diff
@@ -106,20 +106,28 @@ export class ConfigurationServiceLocalStorage<T=any> implements ConfigurationSer
...
         }));
-        
+
         resolve({});
       }).catch(reject);
     });
   }
+
+  createEntry(request: SaveRequest) {
+    return Promise.reject(new Error(messages.notImplemented));
```
Comment (stephanebenayoun) [inline src/provider/LocalStorage/ConfigurationServiceLocalStorage.ts]: We need to implement this two methods for local storage

File: src/provider/LocalStorage/ConfigurationServiceLocalStorage.ts [hunk 3]
```diff
@@ -106,20 +107,51 @@ export class ConfigurationServiceLocalStorage<T=any> implements ConfigurationSer
...
+    return this.changeEntry(request, 'create');
+  };
+
+  public updateEntry(request: SaveRequest<T>) {
+    return this.changeEntry(request, 'update');
+  };
+
+  private changeEntry(request: SaveRequest<T>, mode: 'create' | 'update') {
+    try {
+      validateSaveEntryRequest(request);
```
Comment (stephanebenayoun) [inline src/provider/LocalStorage/ConfigurationServiceLocalStorage.ts:133]:
```suggestion
      validateChangeEntryRequest(request);
```

File: src/provider/Scalecube/ConfigurationServiceScalecube.ts [hunk 4]
```diff
@@ -14,6 +14,7 @@ import {
   SaveResponse,
 } from '../../api';
 import {messages, repositoryKeyRequestValidator, repositoryRequestValidator} from '../../utils';
+import { isNonEmptyString, isObject, validateSaveEntryRequest } from '../../validators';
```
Comment (stephanebenayoun) [inline src/provider/Scalecube/ConfigurationServiceScalecube.ts:17]:
```suggestion
import { isNonEmptyString, isObject, validateChangeEntryRequest } from '../../validators';
```

File: src/provider/Scalecube/ConfigurationServiceScalecube.ts [hunk 5]
```diff
@@ -88,6 +89,25 @@ export class ConfigurationServiceScalecube<T = any> implements ConfigurationServ
...
+    return this.changeEntry(request, 'create');
+  }
+
+  public updateEntry(request: SaveRequest<T>) {
+    return this.changeEntry(request, 'update');
+  }
+
+  private changeEntry = (request: SaveRequest<T>, mode: 'update' | 'create') => {
+    try {
+      validateSaveEntryRequest(request);
```
Comment (stephanebenayoun) [inline src/provider/Scalecube/ConfigurationServiceScalecube.ts:102]:
```suggestion
      validateChangeEntryRequest(request);
```

File: src/validators.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { messages } from './utils';
+
+export const isObject = (val: any) => !!val && typeof val === 'object' && typeof val.map !== 'function';
+
+export const isNonEmptyString = (val: any) => typeof val === 'string' && !!val.trim();
+
+export const validateSaveEntryRequest = (request: any) => {
```
Comment (stephanebenayoun) [inline src/validators.ts:7]:
```suggestion
export const validateChangeEntryRequest = (request: any) => {
```

File: src/provider/Scalecube/ConfigurationServiceScalecube.ts [hunk 7]
```diff
@@ -88,6 +89,25 @@ export class ConfigurationServiceScalecube<T = any> implements ConfigurationServ
...
 
+  public createEntry(request: SaveRequest<T>) {
+    return this.changeEntry(request, 'create');
+  }
+
+  public updateEntry(request: SaveRequest<T>) {
+    return this.changeEntry(request, 'update');
+  }
+
+  private changeEntry = (request: SaveRequest<T>, mode: 'update' | 'create') => {
```
Comment (stephanebenayoun) [inline src/provider/Scalecube/ConfigurationServiceScalecube.ts:100]: Can we extract the `mode` type to API ?

Comment (stephanebenayoun) [reply src/provider/Scalecube/ConfigurationServiceScalecube.ts:100]: not relevant

File: src/provider/LocalStorage/ConfigurationServiceLocalStorage.ts (hunk 3 above)
Comment (dmitriyslaym) [reply src/provider/LocalStorage/ConfigurationServiceLocalStorage.ts:133]: the type name is SaveRequest - the method name was based on this.

**Review by stephanebenayoun — APPROVED** [review]

**Pull request merged**